src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/InsertOperation.java

```diff
@@ -29,16 +30,28 @@ public record InsertOperation(
     CommandContext commandContext,
     List<WritableShreddedDocument> documents,
     boolean ordered,
-    boolean offlineMode)
+    boolean offlineMode,
+    boolean returnDocumentResponses)
     implements ModifyOperation {
+  public record WritableDocAndPosition(int position, WritableShreddedDocument document)
```

Comment:
Make WritableDocAndPosition as a record, because we always want to attach position info to the document?

Reply:
Oh, because you need to sort them later, to make it the the order in the response

Reply:
Right, need to keep track of position to match `documentResponses` indexes. Not cleanest possible way but workable.